sympy/stats/rv.py

```diff
@@ -1017,69 +1022,58 @@ def where(condition, given_condition=None, **kwargs):
     return pspace(condition).where(condition, **kwargs)
 
 
-def sample(expr, condition=None, size=(), **kwargs):
+def sample(expr, condition=None, size=(1,), library='scipy', numsamples=1,
+                                                                    **kwargs):
     """
     A realization of the random expression
 
+    Parameters
+    ==========
+
+    expr : Expression of random variables
+        Expression from which sample is extracted
+    condition : Expr containing RandomSymbols
+        A conditional expression
+    size : int, tuple
+        Represents size of each sample in numsamples
+    library : str
+        - 'scipy' : Sample using scipy
+        - 'numpy' : Sample using numpy
+        - 'pymc3' : Sample using PyMC3
+
+        Choose any of the available options to sample from as string,
+        by default is 'scipy'
+    numsamples : int
+        Number of samples, each with size as ``size``
+
     Examples
     ========
 
     >>> from sympy.stats import Die, sample, Normal
     >>> X, Y, Z = Die('X', 6), Die('Y', 6), Die('Z', 6)
 
-    >>> die_roll = sample(X + Y + Z) # A random realization of three dice
+    >>> die_roll = sample(X + Y + Z) # doctest: +SKIP
     >>> N = Normal('N', 3, 4)
-    >>> samp = sample(N)
-    >>> samp in N.pspace.domain.set
+    >>> samp = next(sample(N))[0] # doctest: +SKIP
+    >>> samp in N.pspace.domain.set # doctest: +SKIP
+    True
+    >>> samp = next(sample(N, N>0))[0] # doctest: +SKIP
+    >>> samp > 0 # doctest: +SKIP
     True
-    >>> samp_list = sample(N, size=4)
-    >>> [sam in N.pspace.domain.set for sam in samp_list]
+    >>> samp_list = next(sample(N, size=4)) # doctest: +SKIP
+    >>> [sam in N.pspace.domain.set for sam in samp_list] # doctest: +SKIP
     [True, True, True, True]
 
-    """
-    return next(sample_iter(expr, condition, size=size, numsamples=1))
+    Returns
+    =======
 
+    sample: iterator object
```

Comment:
I see always an iterator object is returned, so probably a warning can be added for the same.

Reply:
Or adding it in `sample_iter` would be a better idea because every function is using it to generate results, so warning would be raised on any call to it.

Reply:
We need to add a warning of a return type change in only `sample` as others returned iterator object in current master too.

Reply:
Should we add something like this?
```python
>>> from sympy.stats import *
>>> X = Normal('X',1, 2)
>>> itr = sample(X)
/home/smit/Smitlunagariya/sympy/sympy/stats/rv.py:1074: UserWarning: The return type of sample has been changed to return an iterator object.
For more information see https://github.com/sympy/sympy/issues/19061
  warnings.warn("The return type of sample has been changed to return an "
>>> next(itr)
array([1.34687082])
```

Reply:
Looks good.

Reply:
`sample returns an iterator object since version 1.7` would be better though.

Reply:
Update the release notes entry as well before merging about this change.

Reply:
According to the discussions, this looks good to merge now as:
* we have added warnings of change in return type.
* The functions `sample_iter_subs` and `sample_iter_lambdify` are present currently, but I suggest removing them in further versions, as they are of no use now. If you agree with the same then I will open an issue for the removal of them in future

Reply:
> If you agree with the same then I will open an issue for the removal of them in future

That's better. We have a lot of time for `1.7`.